src/asm/lexer.c

```diff
@@ -1089,14 +1091,15 @@ static void readHexNumber(void)
 			c = c - 'A' + 10;
 		else if (c >= '0' && c <= '9')
 			c = c - '0';
+		else if (c == '_' && !empty)
```

Comment:
Why check for `empty` here but not in other functions?

Reply:
In `yylex_NORMAL`, `case '%'` ensures that `secondChar` is 0 or 1 before calling `readBinaryNumber`; `case '&'` ensures that `secondChar` is 0-7 before calling `readNumber(8, 0)`; and there is no `case '_'`, so `default` prevents decimal numbers from beginning with `_`.